src/main/scala/beam/agentsim/infrastructure/ChargingFunctions.scala

```diff
@@ -97,30 +97,66 @@ class ChargingFunctions(
     * @param inquiry ParkingInquiry
     * @return
     */
-  def ifHomeWorkOrLongParkingDurationThenSlowChargingOnly(zone: ParkingZone, inquiry: ParkingInquiry): Boolean = {
-    zone.chargingPointType.forall(chargingPointType =>
-      inquiry.beamVehicle.forall {
-        case vehicle
-            if !vehicle.isRideHail && (isHomeWorkOrOvernight(inquiry) || hasLongParkingDurationButNotCharge(inquiry)) =>
-          !ChargingPointType.isFastCharger(chargingPointType)
-        case _ => true
-      }
-    )
+  def ifHomeWorkOrLongParkingDurationThenSlowChargingOnlyUnlessEnrouting(
```

Comment:
The unless is unnecessary and confuses the overall. How about
```suggestion
  def ifEnrouting_Home_Work_Or_LongParkingDurationThenSlowChargingOnly(
```

Reply:
I think your suggestion implies that if enrouting then slow charging only, which is the opposite behavior.

I added the "UnlessEnrouting" to reflect that the filter now just returns true if we are enrouting, regardless of the other conditions.

Reply:
But isn't that the end functionality anyway? True means to use slow charging unless I am misreading it.

Reply:
This function is part of a filter used to decide if we will allow a particular charging station to be considered as a choice. True means we allow it to be chosen.
when we are enrouting, we only allow fast chargers;
when we are parking for a long time, at home or at work, we only allow slow chargers.

But originally, there was a bug where this filter would return false for everything if we needed to enroute on our way to work or home, because another filter says we need fast chargers, and this would only allow slow ones so they were mutually exclusive.

Basically the true, in this case for enrouting, means we do not care if it is fast or slow because the other filters will take care of it.

Reply:
Perhaps `unlessEnrouting_IfHomeWorkOrLongParkingDurationThenSlowChargingOnly` is a bit clearer? That if you agree that it is the functionality of this.

Reply:
Ultimately this is a really convoluted function. Based on what you said then this should be renamed entirely - to indicate that it selects the charger if true. I broke it down to be a simple boolean expression (feel free to use it if you want - but the `case` might be more readable):

```
lazy val notRideHail_and_homeWorkOvernightOrLongDuration= !vehicle.isRideHail && (isHomeWorkOrOvernight(inquiry) || hasLongParkingDurationButNotCharge(inquiry))
    inquiry.searchMode == ParkingSearchMode.EnRouteCharging || !notRideHail_and_homeWorkOvernightOrLongDuration || 
    	!(notRideHail_and_homeWorkOvernightOrLongDuration && ChargingPointType.isFastCharger(chargingPointType))
```

So, that makes it read to me as super convoluted no matter how I try to word it...so my opinion is to leave the implementation details internal to the method: `allowChargingStationBasedOn(zone, inquiry)`. Otherwise to capture the full implementation it must become something like: `skipChargingStartionIfNotEnroute_notRideHailAndHomeWorkOvernightLongDuration_andIsFastCharger`

Because when I read the implementation further then there is only **1** case where it is false.

Reply:
I'm changing this to `filterBasedOnChargerSpeed`. I think it is a good middle ground between explaining too much (possibly being convoluted) and explaining too little.